pkg/compactor/compactor.go

```diff
@@ -790,10 +792,18 @@ func (c *Compactor) compactUser(ctx context.Context, userID string) error {
 	// out of order chunks or index file too big.
 	noCompactMarkerFilter := compact.NewGatherNoCompactionMarkFilter(ulogger, bucket, c.compactorCfg.MetaSyncConcurrency)
 
+	var blockIDsFetcher block.BlockIDsFetcher
+	if c.storageCfg.BucketStore.BucketIndex.Enabled && c.compactorCfg.BucketIndexBlockIDsFetcherEnalbed {
```

Comment:
I think the goal is to always enable using bucket index to fetch block IDs once this feature is stable and remove this flag totally? Can we add a TODO and mention this?

Reply:
Added a TODO